src/lib.rs

```diff
@@ -309,15 +312,31 @@ impl KVMIntrospectable for KVMi {
     fn get_registers(&self, vcpu: u16) -> Result<(kvm_regs, kvm_sregs, kvm_msrs), Error> {
         let mut regs: kvm_regs = unsafe { mem::MaybeUninit::<kvm_regs>::zeroed().assume_init() };
         let mut sregs: kvm_sregs = unsafe { mem::MaybeUninit::<kvm_sregs>::zeroed().assume_init() };
-        let mut msrs: kvm_msrs = unsafe { mem::MaybeUninit::<kvm_msrs>::zeroed().assume_init() };
+        //let mut msrs: kvm_msrs = unsafe { mem::MaybeUninit::<kvm_msrs>::zeroed().assume_init() };
+        let size = 6;
+        let layout = Layout::from_size_align(
+            mem::size_of::<u32>()
+                + mem::size_of::<u32>()
+                + size * (mem::size_of::<kvm_msr_entry>()),
+            cmp::max(mem::align_of::<u32>(), mem::align_of::<kvm_msr_entry>()),
+        )
+        .unwrap();
+        let msrs = alloc(layout) as *mut kvm_msrs;
+        let mut msr_entry_slice: &[kvm_msr_entry] =
+            unsafe { slice::from_raw_parts(unsafe { &mut (*msrs).entries }, 6) };
```

Comment:
```suggestion
        let msr_entry_slice: &mut [kvm_msr_entry] =
            unsafe { slice::from_raw_parts_mut( (*msrs).entries.as_mut_ptr() as *mut _, 6) };
```

Reply:
Consider adding this as a method that operates on `kvm_msrs` so you can reuse it in libmicrovmi.